### PR TITLE
81721 FT-EES Scenario verification with suspend & withdrawn 

### PR DESCRIPTION
**Pull request by prashantmgupta**

### CODE REVIEW

File: tests/UKHO.ERPFacade.API.FunctionalTests/ERPFacadeScenarioData/Scenario Mapping.xlsx
Comment (Sadha1501493) [inline tests/UKHO.ERPFacade.API.FunctionalTests/ERPFacadeScenarioData/Scenario Mapping.xlsx:1]: Please update 'Covered Business' column for appended scenarios.

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs [hunk 1]
```diff
@@ -573,7 +625,7 @@ private static bool verifyCreateENCCell(string childCell, Item item)
                     if (!item.PRODTYPE.Equals(product.ProductType[4..]))
                         AttrNotMatched.Add(nameof(item.PRODTYPE));
                     //if (!item.PRODUCTNAME.Equals(product.ProductName))(!product.InUnitsOfSale.Contains(item.PRODUCTNAME)) && (!getUoSInfo(item.PRODUCTNAME).UnitType.Contains("Unit"))
```
Comment (Sadha1501493) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs:627]: Please removed unused code.

**Review by Sadha1501493 — CHANGES_REQUESTED** [review]

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs [hunk 2]
```diff
@@ -227,8 +271,16 @@ private static bool getUoSUnitType(string pRODUCTNAME)
                         AttrNotMatched.Add(nameof(item.PRODUCT));
                     if (!item.PRODTYPE.Equals(product.ProductType[4..]))
                         AttrNotMatched.Add(nameof(item.PRODTYPE));
-                    if (!item.PRODUCTNAME.Equals(product.ProductName))
-                        AttrNotMatched.Add(nameof(item.PRODUCTNAME));
+
+                    //if (!item.PRODUCTNAME.Equals(product.ProductName))
+                    //   AttrNotMatched.Add(nameof(item.PRODUCTNAME));
```
[the remark this answers is not in the document]
Comment (prashantmgupta) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs:276]: Resolved

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs [hunk 3]
```diff
@@ -227,8 +271,16 @@ private static bool getUoSUnitType(string pRODUCTNAME)
...
+
+                    if ((!product.InUnitsOfSale.Contains(item.PRODUCTNAME)) && (!getUoSInfo(item.PRODUCTNAME).UnitOfSaleType.Contains("unit")))
+                       AttrNotMatched.Add(nameof(item.PRODUCTNAME));
+
+                    //if (!item.PRODUCTNAME.Equals(getUnitOfSalesWithTypeEqualsUnit(product.ProductName).UnitName))
+                    //  AttrNotMatched.Add(nameof(item.PRODUCTNAME));
+
                     if (!item.AGENCY.Equals(product.Agency))
```
[the remark this answers is not in the document]
Comment (prashantmgupta) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs:284]: Resolved

File: tests/UKHO.ERPFacade.API.FunctionalTests/ERPFacadeScenarioData/Scenario Mapping.xlsx
Comment (prashantmgupta) [reply tests/UKHO.ERPFacade.API.FunctionalTests/ERPFacadeScenarioData/Scenario Mapping.xlsx:1]: Updated

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs (hunk 1 above)
Comment (prashantmgupta) [reply tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/SAPXmlHelper.cs:627]: Resolved

**Review by jrippington — APPROVED** [review]
Looks good

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Review by RaviK2233 — APPROVED** [review]
Looks good

**Pull request merged**